bookkeeper-server/src/main/java/org/apache/bookkeeper/util/ZkUtils.java

```diff
@@ -134,8 +134,21 @@ public static void asyncDeleteFullPathOptimistic(final ZooKeeper zk, final Strin
             public void processResult(int rc, String path, Object ctx) {
                 if (rc == Code.OK.intValue()) {
                     String parent = new File(originalPath).getParent().replace("\\", "/");
-                    asyncDeleteFullPathOptimistic(zk, parent, -1, callback, leafNodePath);
+                    zk.getData(parent, false, (dRc, dPath, dCtx, data, stat) -> {
+                        if (Code.OK.intValue() == dRc && (stat != null && stat.getNumChildren() == 0)) {
+                            asyncDeleteFullPathOptimistic(zk, parent, -1, callback, leafNodePath);
+                        } else {
+                            // parent node is not empty so, complete the
+                            // callback
+                            if (path.equals(leafNodePath)) {
```

Comment:
wondering what is the difference between the if/else cases here? since rc is OK here, will it not be the same?

Reply:
@reddycharan  that's true.. rc will be always OK here. fixed it.